src/lm_buddy/jobs/_entrypoints/ragas.py

```diff
@@ -0,0 +1,105 @@
+from pathlib import Path
+
+from datasets import load_dataset
+from langchain_community.embeddings import HuggingFaceEmbeddings
+from langchain_openai import ChatOpenAI
+from lmbuddy.integrations.huggingface import HuggingFaceAssetLoader
+from ragas import evaluate
+from ragas.metrics import answer_relevancy, context_precision, context_recall, faithfulness
+
+from lm_buddy.integrations.wandb import artifact_loader
+from lm_buddy.integrations.wandb.artifact_config import WandbArtifactConfig
+from lm_buddy.integrations.wandb.artifact_utils import (
+    ArtifactType,
+    build_directory_artifact,
+    default_artifact_name,
+)
+from lm_buddy.integrations.wandb.run_utils import wandb_init_from_config
+from lm_buddy.jobs.common import EvaluationResult, LMBuddyJobType
+from lm_buddy.jobs.configs import RagasJobConfig
+
+RAGAS_METRICS_MAP = {
+    "faithfulness": faithfulness,
+    "answer_relevancy": answer_relevancy,
+    "context_recall": context_recall,
+    "context_precision": context_precision,
+}
+
+
+def run_eval(config: RagasJobConfig, artifact_loader: artifact_loader) -> Path:
+    # load dataset from W&B artifact
+    hf_loader = HuggingFaceAssetLoader(artifact_loader)
+    evaluation_dataset = hf_loader.load_dataset(config.dataset)
+
+    # ragas custom model args
+    ragas_args = {}
+
+    # load embedding model
+    ragas_args["embeddings"] = HuggingFaceEmbeddings(model_name=config.evaluation.embedding_model)
+
+    # configure ragas to point to vllm instance for generation
+    ragas_args["llm"] = ChatOpenAI(
```

Comment:
Are we able to bypass langchain here and use the vllm implementation since we already have access? https://github.com/vllm-project/vllm/blob/f721096d48a7e3b98dffcb9b400bf58989cef64d/examples/openai_completion_client.py#L7. worth checking https://api.python.langchain.com/en/latest/language_models/langchain_core.language_models.base.BaseLanguageModel.html

Reply:
We do have a (barebones) OpenAI client implementation in the prometheus entrypoint, perhaps it might help if we moved it to a common library and reuse it?

Reply:
I'm looking into how ragas is calling the completions in each evaluate method to see if it will work

Reply:
It looks like ragas basically uses langchain under the hood to load the models https://github.com/explodinggradients/ragas/blob/962d40d1f5f45fe57266ec35176329b07dfc0ae5/src/ragas/adaptation.py#L3, and it doesn't have direct overrides for vllm (supporting it essentially through langchain) so I think we will have to keep this dependency for now. They might add native support for it later, or we can contribute back to it and then refactor this out after the fact if it makes sense.